fastsim-core/benches/interp_bench.rs

```diff
@@ -0,0 +1,195 @@
+        values: ArrayD::from_shape_vec(IxDyn(&[100, 100]), values_data).unwrap(),
+    });
+    interp_2d_multi.validate().unwrap();
+    // Sample 1,000 points
+    let points: Vec<Vec<f64>> = (0..1_000)
+        .map(|_| vec![rng.gen::<f64>() * 99., rng.gen::<f64>() * 99.])
+        .collect();
+    for point in points {
+        interp_2d_multi
+            .interpolate(&point, &Strategy::Linear)
+            .unwrap();
+    }
+}
+
+/// 3-D interpolation (hardcoded)
+fn benchmark_3D() {
+    let seed = 1234567890;
+    let mut rng = StdRng::seed_from_u64(seed);
+    let grid_data: Vec<f64> = (0..100).map(|x| x as f64).collect();
+    // Generate interpolator data (same as N-D benchmark) and arrange into `Vec<Vec<Vec<f64>>>`
+    let values_data: Vec<f64> = (0..1_000_000).map(|_| rng.gen::<f64>()).collect();
+    let values_data: Vec<Vec<Vec<f64>>> = (0..100)
+        .map(|x| {
+            (0..100)
+                .map(|y| values_data[(100 * (y + 100 * x))..(100 + 100 * (y + 100 * x))].into())
+                .collect()
+        })
+        .collect();
+    // Create a 3-D interpolator with 100x100x100 data (1,000,000 points)
+    let interp_3d = Interpolator::Interp3D(Interp3D {
+        x: grid_data.clone(),
+        y: grid_data.clone(),
+        z: grid_data.clone(),
+        f_xyz: values_data,
+    });
+    interp_3d.validate().unwrap();
+    // Sample 1,000 points
+    let points: Vec<Vec<f64>> = (0..1_000)
+        .map(|_| {
+            vec![
+                rng.gen::<f64>() * 99.,
+                rng.gen::<f64>() * 99.,
+                rng.gen::<f64>() * 99.,
+            ]
+        })
+        .collect();
+    for point in points {
+        interp_3d.interpolate(&point, &Strategy::Linear).unwrap();
+    }
+}
+
+/// 3-D interpolation (multilinear interpolator)
+fn benchmark_3D_multi() {
+    let seed = 1234567890;
+    let mut rng = StdRng::seed_from_u64(seed);
+    // Generate interpolator data (same as hardcoded benchmark)
+    let grid_data: Vec<f64> = (0..100).map(|x| x as f64).collect();
+    let values_data: Vec<f64> = (0..1_000_000).map(|_| rng.gen::<f64>()).collect();
+    // Create an N-D interpolator with 100x100x100 data (1,000,000 points)
+    let interp_3d_multi = Interpolator::InterpND(InterpND {
+        grid: vec![grid_data.clone(), grid_data.clone(), grid_data.clone()],
+        values: ArrayD::from_shape_vec(IxDyn(&[100, 100, 100]), values_data).unwrap(),
+    });
+    interp_3d_multi.validate().unwrap();
+    // Sample 1,000 points
+    let points: Vec<Vec<f64>> = (0..1_000)
+        .map(|_| {
+            vec![
+                rng.gen::<f64>() * 99.,
+                rng.gen::<f64>() * 99.,
+                rng.gen::<f64>() * 99.,
+            ]
+        })
+        .collect();
+    for point in points {
+        interp_3d_multi
+            .interpolate(&point, &Strategy::Linear)
+            .unwrap();
+    }
+}
+
+pub fn criterion_benchmark(c: &mut Criterion) {
```

Comment:
@kylecarow , could you add a doc string here to explain how to trigger the benchmarks?